action/protocol/staking/protocol.go

```diff
@@ -109,3 +121,31 @@ func (p *Protocol) Register(r *protocol.Registry) error {
 func (p *Protocol) ForceRegister(r *protocol.Registry) error {
 	return r.ForceRegister(protocolID, p)
 }
+
+// CreateGenesisStates creates genesis states
+func (p *Protocol) CreateGenesisStates(ctx context.Context, sm protocol.StateManager) error {
+	if p.inMemCandidates.Size() > 0 {
+		return nil
+	}
+
+	// read all candidates from stateDB
+	stateDB := sm.GetDB()
+	candPkHash, candValue, err := stateDB.Filter(factory.CandidateNameSpace, func(k, v []byte) bool {
+		return bytes.HasPrefix(k, []byte{candidate})
+	})
```

Comment:
Let me think a way to not add back getdb

Reply:
stateDB contains an underlying KVStore, i think it is OK to expose that for use

suggest we not spend time/effort on finding a way to not add back getdb